pipe-cli/src/utilities/datastorage_operations.py

```diff
@@ -141,6 +121,65 @@ def cp(cls, source, destination, recursive, force, exclude, include, quiet, tags
             click.echo('Error: %s' % str(error), err=True)
             sys.exit(1)
 
+    @classmethod
+    def _filter_items(cls, items, manager, source_wrapper, destination_wrapper, permission_to_check,
+                      include, exclude, force, quiet, skip_existing):
+        filtered_items = []
+        for item in items:
+            full_path = item[1]
+            relative_path = item[2]
+            size = item[3]
+
+            if relative_path.endswith(FOLDER_MARKER):
+                filtered_items.append(item)
+                continue
+
+            # check that we have corresponding permission for the file before take action
+            if source_wrapper.is_local() and not os.access(full_path, permission_to_check):
+                continue
+            if not include and not exclude:
+                if source_wrapper.is_file() and not source_wrapper.path == full_path:
+                    continue
+                if not source_wrapper.is_file():
+                    possible_folder_name = source_wrapper.path_with_trailing_separator()
+                    if not full_path.startswith(possible_folder_name):
+                        continue
+            if not PatternMatcher.match_any(relative_path, include):
+                if not quiet:
+                    click.echo("Skipping file {} since it doesn't match any of include patterns [{}]."
+                               .format(full_path, ",".join(include)))
+                continue
+            if PatternMatcher.match_any(relative_path, exclude, default=False):
+                if not quiet:
+                    click.echo("Skipping file {} since it matches exclude patterns [{}]."
+                               .format(full_path, ",".join(exclude)))
+                continue
+
+            if force and not skip_existing:
+                filtered_items.append(item)
+                continue
+
+            destination_key = manager.get_destination_key(destination_wrapper, relative_path)
+            destination_size = manager.get_destination_size(destination_wrapper, destination_key)
+            destination_is_empty = destination_size is None
+            if destination_is_empty:
+                filtered_items.append(item)
+                continue
+            if skip_existing:
+                source_key = manager.get_source_key(source_wrapper, full_path)
+                source_size = manager.get_source_size(source_wrapper, source_key, size)
```

Comment:
It seems like we can just use `size` here without `manager.get_source_size` method at all.

Reply:
done